### PR TITLE
[#169425266] Update authentication state on logout

### PR DESCRIPTION
**Pull request by Edigio**
The authentication status is updated on the logout

### CODE REVIEW

**Comment by digitalcitizenship** [comment]
## Affected stories

  * 🌟 [#169425266](https://www.pivotaltracker.com/story/show/169425266): Quando parte una richiesta di logout deve essere aggiornato lo stato authentication e non lo stato del componente SelectLogoutOption

<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: .circleci/config.yml [hunk 1]
```diff
@@ -7,7 +7,7 @@
 defaults: &defaults
   working_directory: ~/italia-app
 
-node_cache_key: &node_cache_key v1.10-dependencies-{{ checksum "yarn.lock" }}
+node_cache_key: &node_cache_key v1.11-dependencies-{{ checksum "yarn.lock" }}
```
Comment (Undermaken) [inline .circleci/config.yml]: I merged [this PR](https://github.com/teamdigitale/io-app/pull/1514) to solve the problem 
You can revert this change

File: ts/components/SelectLogoutOption.tsx [hunk 2]
```diff
@@ -114,11 +100,21 @@ class SelectLogoutOption extends React.PureComponent<Props, State> {
     BackHandler.removeEventListener("hardwareBackPress", this.onBackPress);
   }
 }
+
+const mapStateToProps = (state: GlobalState) => {
+  return {
+    isLogoutInitialized:
```
Comment (Undermaken) [inline ts/components/SelectLogoutOption.tsx]:
```suggestion
    isLogoutInProgress:
```

File: ts/store/reducers/authentication.ts [hunk 3]
```diff
@@ -176,6 +182,13 @@ const reducer = (
     };
   }
 
+  if (isActionOf(logoutRequest, action) && isLoggedIn(state)) {
```
Comment (Undermaken) [inline ts/store/reducers/authentication.ts]: Perhaps did you miss to implement the **"COMPLETE"** case?

File: ts/store/reducers/authentication.ts [hunk 4]
```diff
@@ -39,13 +42,15 @@ export type LoggedInWithoutSessionInfo = Readonly<{
...
   sessionToken: SessionToken;
+  logoutState: LogoutState;
 }>;
 
 // The user is logged in and we also have all session info
 export type LoggedInWithSessionInfo = Readonly<{
   kind: "LoggedInWithSessionInfo";
   idp: IdentityProvider;
   sessionToken: SessionToken;
+  logoutState: LogoutState;
```
Comment (Undermaken) [inline ts/store/reducers/authentication.ts]: Is it needed?
Could we just add another kind? i.e **_LogoutRequested_**
cc @valentinacala

Comment (Edigio) [reply ts/store/reducers/authentication.ts]: mmm yes i added another kind, I think it's the right solution
